grunt_tasks/plugin.js

```diff
@@ -401,13 +391,23 @@ module.exports = function (grunt) {
                         library: `girder_plugin_${plugin}`
                     },
                     resolve: {
+                        alias: {
+                            // Make sure we use the core jquery rather than pulling one
+                            // from a plugin's local node modules. Multiple jquery versions
+                            // breaks due to use of side effects for jquery plugins.
+                            'jquery': path.resolve(paths.node_modules, 'jquery')
+                        },
                         modules: [
-                            path.resolve(dir, 'node_modules')
+                            path.resolve(dir, 'node_modules'),
+                            pluginNodeDir,
+                            paths.node_modules
```

Comment:
So it looks like the search order is:

1. `plugins\<plugin_name>\node_modules` which exists if it is a plugin that builds independently with its own `node_modules`.
2. `node_modules\girder_plugin_<plugin_name>\node_modules` which exists if it is a plugin with a `package.json` that builds with the Girder build.
3. Girder's `node_modules` if it is a standard plugin without a `package.json`.

Is this right? Could we put a bit of documentation up to this effect?

Reply:
> 1. plugins\<plugin_name>\node_modules which exists if it is a plugin that builds independently with its own node_modules.

This is correct.

> 2. node_modules\girder_plugin_<plugin_name>\node_modules which exists if it is a plugin with a package.json that builds with the Girder build.

My understanding is that this is the location used if a plugin sets `localNodeModules: true`, and the location itself is a bit of an internal detail, the rationale of which I don't fully understand at this time.

> Girder's node_modules if it is a standard plugin without a package.json.

This fallback path will be used regardless of whether the plugin has its own local node_modules, in case it needs to pull in something from the global node_modules. (That's probably what you meant, just wanted to clarify.)

Thinking about it a little more, I think I will actually switch the order of 1 and 2 -- I think in the case where a plugin has `localNodeModules: true` and also has its own `node_modules` directory that was installed for whatever reason, during the girder plugin build, it should pull from the `localNodeModules` stuff first, since that is definitely a girder plugin directive, and we don't know the reason `node_modules` is there.

I will add documentation about what we are guaranteeing, and how to alter it using webpack.helper.js. I'll leave off the specifics of the path used in the case of `localNodeModules: true`, since that is an implementation detail that may even change in the future.